src/doom/p_setup.c

```diff
@@ -531,6 +531,10 @@ void P_LoadBlockMap (int lump)
 
     lumplen = W_LumpLength(lump);
     count = lumplen / 2;
+    if(count >= 32766)
+    {
+	I_Error("Blockmap too large");
```

Comment:
This is not a good error message. A better error message would include the two values that were compared and a link to the doomwiki.org page about the static limit.